fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4

```diff
@@ -135,6 +136,7 @@ HLL_UNION: 'HLL_UNION';
 HOUR: 'HOUR';
 IDENTIFIED: 'IDENTIFIED';
 IF: 'IF';
+IMPERSONATE: 'IMPERSONATE';
```

Comment:
should make new added keywords nonReserve

Reply:
and I have a question, is it necessary to make all-priv a keyword?
How about `GRANT literal` and when analyzing, this literal will be checked if it is a valid privilege.

Reply:
and I have a question, is it necessary to make all-priv a keyword?
How about `GRANT literal` and when analyzing, this literal will be checked if it is a valid privilege.

Reply:
1. Impersonate privilege is not like other privileges that can specify several privileges in one statement. You can't use `GRANT IMPERSONATE, SELECT ON XX TO XX`, but you can use `GRANT SELECT, LOAD ON XX TO XX`. I also refer to the Mysql document and found that the `PROXY` privilege is put aside as a single rule.
2. We do not use a keyword to explicitly define a securable entity like SQL Server, instead, we add a lot of if/else to check whether the securable entity is table or DB, or global. To put impersonate privilege in the original statement, I'll have to make it more complex to read. I think it's best to add a new rule and let PM decide later if we should refactor the SQL syntax.
3. Currently, the GRANT statement is under the old analyzer framework, I think it's better to add new logic to the new analyzer framework, but I do not think we should move the old statement to the new framework right now since we're about to refactor the whole framework later.

For the above reasons, I add a new keyword and a new rule.